test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1424,4 +1424,243 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
       });
     });
   });
+
+  context('14. Decryption Events', function () {
+    let setupClient;
+    let clientEncryption;
+    let keyId;
+    let cipherText;
+    let malformedCiphertext;
+    let encryptedClient;
+    let aggregateSucceeded;
+    let aggregateStarted;
+    let aggregateFailed;
+
+    beforeEach(async function () {
+      const mongodbClientEncryption = this.configuration.mongodbClientEncryption;
+      // Create a MongoClient named ``setupClient``.
+      setupClient = this.configuration.newClient();
+      // Drop and create the collection ``db.decryption_events``.
+      const db = setupClient.db('db');
+      await db.dropCollection('decryption_events').catch(error => {
+        if (!error.message.match(/ns not found/)) {
+          throw error;
+        }
+      });
+      await db.createCollection('decryption_events');
+      // Create a ClientEncryption object named ``clientEncryption`` with these options:
+      //   ClientEncryptionOpts {
+      //     keyVaultClient: <setupClient>,
+      //     keyVaultNamespace: "keyvault.datakeys",
+      //     kmsProviders: { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+      //   }
+      clientEncryption = new mongodbClientEncryption.ClientEncryption(setupClient, {
+        keyVaultNamespace: 'keyvault.datakeys',
+        kmsProviders: getKmsProviders(LOCAL_KEY),
+        bson: BSON
+      });
+      // Create a data key with the "local" KMS provider.
+      // Storing the result in a variable named ``keyID``.
+      keyId = await clientEncryption.createDataKey('local');
+      // Use ``clientEncryption`` to encrypt the string "hello" with the following ``EncryptOpts``:
+      //   EncryptOpts {
+      //     keyId: <keyID>,
+      //     algorithm: "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+      //   }
+      // Store the result in a variable named ``ciphertext``.
+      cipherText = await clientEncryption.encrypt('hello', {
+        keyId: keyId,
+        algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic'
+      });
+      // Copy ``ciphertext`` into a variable named ``malformedCiphertext``.
+      // Change the last byte to 0. This will produce an invalid HMAC tag.
+      const buffer = Buffer.from(cipherText.buffer);
+      buffer.writeInt8(0, buffer.length - 1);
+      malformedCiphertext = new Binary(buffer, 6);
+      // Create a MongoClient named ``encryptedClient`` with these ``AutoEncryptionOpts``:
+      //   AutoEncryptionOpts {
+      //     keyVaultNamespace: "keyvault.datakeys";
+      //     kmsProviders: { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+      //   }
+      // Configure ``encryptedClient`` with "retryReads=false".
+      encryptedClient = this.configuration.newClient(
+        {},
+        {
+          retryReads: false,
```

Comment:
question: is there a reason that retryReads is set to false here?  I don't see that listed as a part of the prose test setup.  Is this just related to FLE in general?

Reply:
nevermind, it was the next line in the test setup 😅